commit f9929ef6a2a55f03aac61248c6a3a987b8546f2a
Author: Namjae Jeon <linkinjeon@kernel.org>
Date:   Tue Feb 1 18:20:34 2022 +0900

    ksmbd: add support for key exchange
    
    When mounting cifs client, can see the following warning message.
    
    CIFS: decode_ntlmssp_challenge: authentication has been weakened as server
    does not support key exchange
    
    To remove this warning message, Add support for key exchange feature to
    ksmbd. This patch decrypts 16-byte ciphertext value sent by the client
    using RC4 with session key. The decrypted value is the recovered secondary
    key that will use instead of the session key for signing and sealing.
    
    Signed-off-by: Namjae Jeon <linkinjeon@kernel.org>
    Signed-off-by: Steve French <stfrench@microsoft.com>

diff --git a/fs/Kconfig b/fs/Kconfig
index 7a2b11c0b803..6c7dc1387beb 100644
--- a/fs/Kconfig
+++ b/fs/Kconfig
@@ -369,8 +369,8 @@ source "fs/ksmbd/Kconfig"
 
 config SMBFS_COMMON
 	tristate
-	default y if CIFS=y
-	default m if CIFS=m
+	default y if CIFS=y || SMB_SERVER=y
+	default m if CIFS=m || SMB_SERVER=m
 
 source "fs/coda/Kconfig"
 source "fs/afs/Kconfig"